src/transport/mqtt/connection.rs

```diff
@@ -283,12 +280,36 @@ struct Connection {
     //       https://doc.rust-lang.org/std/sync/struct.Exclusive.html
     eventloop: SyncWrapper<EventLoop>,
     provider: TransportProvider,
+    // Wether the stored introspection matches the current one
+    session_sync: bool,
 }
 
 impl Connection {
     fn eventloop_mut(&mut self) -> &mut EventLoop {
         self.eventloop.get_mut()
     }
+
+    async fn update_session_status<S>(&mut self, interfaces: &Interfaces, store: &StoreWrapper<S>)
```

Comment:
I would make this function return the session status instead of changing it, so we make sure it's false in the error case and early returns
```suggestion
    #[must_use]
    async fn update_session_status<S>(&mut self, interfaces: &Interfaces, store: &StoreWrapper<S>) -> bool
```